deploy/lib/server_config.rb

```diff
@@ -947,16 +945,24 @@ def capture
     if target_db != nil
       tmp_dir = Dir.mktmpdir
       logger.debug "using temp dir " + tmp_dir
-      logger.info "Retrieving source and REST config from #{target_db}..."
 
-      save_files_to_fs(target_db, "#{tmp_dir}/src")
+      if (port != nil)
+        logger.info "Retrieving source and REST config from #{target_db}..."
+      else
+        logger.info "Retrieving source from #{target_db}..."
+      end
 
-      # set up the options
-      FileUtils.cp_r(
-        "#{tmp_dir}/src/#{@properties['ml.group']}/" + target_db.sub("-modules", "") + "/rest-api/.",
-        @properties['ml.rest-options.dir']
-      )
-      FileUtils.rm_rf("#{tmp_dir}/src/#{@properties['ml.group']}/")
+      # send the target db, the destination directory, and port to know if REST for ML7+
+      save_files_to_fs(target_db, "#{tmp_dir}/src", port)
+
+      if (port != nil)
+        # set up the options
+        FileUtils.cp_r(
```

Comment:
I like throwing the error idea, that way the user does not end up with something unexpected. I have implemented this and will be committing momentarily. I will note that it also fixes issue #298 since "capture" will no longer try to save REST configuration if the folder does not exists in the project.